File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/api/RemoteDataSource.java
package com.mondiamedia.ahmedbadr.githubreopos.api;

import javax.inject.Inject;

public class RemoteDataSource {

    private RepositoriesService repositoriesService;

    @Inject
    public RemoteDataSource(RepositoriesService repositoriesService) {
        this.repositoriesService = repositoriesService;
    }

    public RepositoriesService createRepositoriesService() {
        return repositoriesService;
    }
}

File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/api/RemoteDataSource.kt
package com.mondiamedia.ahmedbadr.githubreopos.api

import javax.inject.Inject

class RemoteDataSource @Inject
constructor(private val repositoriesService: RepositoriesService) {

    fun createRepositoriesService(): RepositoriesService {
        return repositoriesService
    }
}
